### PR TITLE
allow for null booolean and datetime fields

### PR DESCRIPTION
**Pull request by fredkingham**

### CODE REVIEW

**Comment by davidmiller** [comment]
+10 moving as much of the ./bin/opal stuff into opal.core as possible...

File: opal/core/scaffold.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    Given a RECORD, return it's relative template dir
+    """
+    modelsfile = inspect.getfile(record)
+    if modelsfile.endswith('.pyc'):
+        modelsfile = modelsfile.replace('.pyc', '.py')
+    appdir = ffs.Path(modelsfile)[:-1]
+    templates = appdir/'templates'
+    return templates
+
+def create_template(record, scaffold_base, template_type):
```
Comment (davidmiller) [inline opal/core/scaffold.py]: Is this function used anywhere?

**Comment by fredkingham** [comment]
I think its used by create_display_template_for and create_form_template_for

I think there maybe some cleaning up we could do to remove duplication between the functions but best to get them all in at least a testable place

**Comment by davidmiller** [comment]
Sorry - meant the `def create_template(record, scaffold_base, template_type):` one below - it's just a pass?

**Comment by fredkingham** [comment]
are sorry, that feels like something I started doing as a refactor, then
figured that it'd be best to just get everything copied across and unit
tested before any further steps.

(Not sure a refactor is a priority, its all pretty good)

**Pull request merged**